### PR TITLE
[Bug Fix] Fix issue with #suspend

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Description
- Described by Dracaena of EZ Server [here](https://discord.com/channels/212663220849213441/1240979738366054451).

## Type of Change
- [X] Bug fix

# Testing
image
image
image
image

# Testing (Post-Repository Change)
image
image
image
image

# Checklist
- [X] I have tested my changes
- [X] I have performed a self-review of my code. Ensuring variables, functions and methods are named in a human-readable way, comments are added only where naming of variables, functions and methods can't give enough context.
- [X] I own the changes of my code and take responsibility for the potential issues that occur

### CODE REVIEW

File: common/repositories/character_data_repository.h [hunk 1]
```diff
@@ -64,6 +64,134 @@ class CharacterDataRepository: public BaseCharacterDataRepository {
...
+			e.zone_id                 = row[6] ? static_cast<uint32_t>(strtoul(row[6], nullptr, 10)) : 0;
+			e.zone_instance           = row[7] ? static_cast<uint32_t>(strtoul(row[7], nullptr, 10)) : 0;
+			e.y                       = row[8] ? strtof(row[8], nullptr) : 0;
+			e.x                       = row[9] ? strtof(row[9], nullptr) : 0;
+			e.z                       = row[10] ? strtof(row[10], nullptr) : 0;
+			e.heading                 = row[11] ? strtof(row[11], nullptr) : 0;
+			e.gender                  = row[12] ? static_cast<uint8_t>(strtoul(row[12], nullptr, 10)) : 0;
+			e.race                    = row[13] ? static_cast<uint16_t>(strtoul(row[13], nullptr, 10)) : 0;
+			e.class_                  = row[14] ? static_cast<uint8_t>(strtoul(row[14], nullptr, 10)) : 0;
+			e.level                   = row[15] ? static_cast<uint32_t>(strtoul(row[15], nullptr, 10)) : 0;
```
Comment (Akkadius) [inline common/repositories/character_data_repository.h:100]: We don't want to be defining all of this manually and should be leaving it to code generation.

We'd be better off having a `GetWhere` here, return a vector/array of struct entries and just return the first entry and remove all of this

**Comment by Akkadius** [comment]
Minor comment should be addressed, will need some quick testing otherwise looks good

File: common/database.cpp [hunk 2]
```diff
@@ -2094,3 +2094,28 @@ void Database::ClearGuildOnlineStatus()
...
+	}
+
+	const int replaced = AccountRepository::ReplaceMany(*this, l);
+
+	if (!replaced) {
+		LogError(
+			"Failed to clear expired suspensions for [{}] account{}",
+			l.size(),
+			l.size() != 1 ? "s" : ""
+		);
```
[the remark this answers is not in the document]
Comment (Kinglykrab) [reply common/database.cpp:2119]: True enough. I’ll remove it.

File: world/main.cpp [hunk 3]
```diff
@@ -179,6 +179,8 @@ int main(int argc, char **argv)
 	EQTimeTimer.Start(600000);
 	Timer parcel_prune_timer(86400000);
 	parcel_prune_timer.Start(86400000);
+	Timer expired_suspensions_prune_timer(60000);
+	expired_suspensions_prune_timer.Start(60000);
```
Comment (Akkadius) [inline world/main.cpp:183]: Does this need to run every minute ?

Once an hour is fine. 

We'll also need to make sure when we're querying for the account during login we are filtering by suspension date as well. That way suspensions are cleared the moment time passes, not when we null out the field.

Comment (Kinglykrab) [reply world/main.cpp]: Removed timer per our discussion.

**Review by Akkadius — APPROVED** [review]

**Pull request merged**